File: app/src/main/java/com/skybox/seven/covid/epoxy/contacts/ContactsController.java
package com.skybox.seven.covid.epoxy.contacts;

import com.airbnb.epoxy.AutoModel;
import com.airbnb.epoxy.Typed2EpoxyController;
import com.skybox.seven.covid.epoxy.generic.ListLoadingModel_;
import com.skybox.seven.covid.model.ContactModel;

import java.util.List;

public class ContactsController extends Typed2EpoxyController<Boolean,List<ContactModel.ContactUsersContacts>> {
    @AutoModel
    ListLoadingModel_ loadingModel_;

    @Override
    protected void buildModels(Boolean loading,List<ContactModel.ContactUsersContacts> contacts) {
        loadingModel_.addIf(loading, this);
        for (ContactModel.ContactUsersContacts contact: contacts) {
            new ContactEpoxyModel_().id(contact.getUser().getPhone()).contact(contact).addTo(this);
        }
    }

}

File: app/src/main/java/com/skybox/seven/covid/epoxy/contacts/ContactsController.kt
package com.skybox.seven.covid.epoxy.contacts

import com.airbnb.epoxy.Typed2EpoxyController
import com.skybox.seven.covid.epoxy.generic.ListLoadingModel_
import com.skybox.seven.covid.model.ContactModel.ContactUsersContacts

class ContactsController : Typed2EpoxyController<Boolean, List<ContactUsersContacts>?>() {
    override fun buildModels(loading: Boolean, contacts: List<ContactUsersContacts>?) {
        ListLoadingModel_().id("loading").addIf(loading, this)
        if (contacts != null) {
            contacts.forEach {
                ContactEpoxyModel_().id(it.user.phone).contact(it).addTo(this)
            }
        } else {

        }
    }
}